Library/ViewModels/RewardCellViewModel.swift

```diff
@@ -0,0 +1,215 @@
+import KsApi
+import Prelude
+import ReactiveSwift
+
+public protocol RewardCellViewModelInputs {
+  func configureWith(project: Project, rewardOrBacking: Either<Reward, Backing>)
+  func pledgeButtonTapped()
+  func rewardCardTapped()
+}
+
+public protocol RewardCellViewModelOutputs {
+  var cardUserInteractionIsEnabled: Signal<Bool, Never> { get }
+  var conversionLabelHidden: Signal<Bool, Never> { get }
+  var conversionLabelText: Signal<String, Never> { get }
+  var descriptionLabelText: Signal<String, Never> { get }
+  var items: Signal<[String], Never> { get }
+  var includedItemsStackViewHidden: Signal<Bool, Never> { get }
+  var pledgeButtonEnabled: Signal<Bool, Never> { get }
+  var pledgeButtonTitleText: Signal<String, Never> { get }
+  var rewardMinimumLabelText: Signal<String, Never> { get }
+  var rewardSelected: Signal<Int, Never> { get }
+  var rewardTitleLabelHidden: Signal<Bool, Never> { get }
+  var rewardTitleLabelText: Signal<String, Never> { get }
+}
+
+public protocol RewardCellViewModelType {
+  var inputs: RewardCellViewModelInputs { get }
+  var outputs: RewardCellViewModelOutputs { get }
+}
+
+public final class RewardCellViewModel: RewardCellViewModelType, RewardCellViewModelInputs,
+  RewardCellViewModelOutputs {
+  public init() {
+    let projectAndRewardOrBacking: Signal<(Project, Either<Reward, Backing>), Never> =
+      self.projectAndRewardOrBackingProperty.signal.skipNil()
+
+    let project: Signal<Project, Never> = projectAndRewardOrBacking.map(first)
+
+    let reward: Signal<Reward, Never> = projectAndRewardOrBacking
+      .map { project, rewardOrBacking -> Reward in
+        rewardOrBacking.left
+          ?? rewardOrBacking.right?.reward
```

Comment:
Am I the only one really confused about what's going on in here?

Reply:
I think this is how it was written before in what is now `DeprecatedRewardCellViewModel.swift` - https://github.com/kickstarter/ios-oss/pull/700/files#diff-be538dae7b71e33a0e267c88f43d2024R52. Agreed the coalescing makes is a little hard to read but I think @ifbarrera didn't want to make any changes to how the code works as far as possible.